docs/hugo/content/architecture-decision-records/ADR-2022-01-Reconciler-Extensions.md

```diff
@@ -72,26 +74,26 @@ type ErrorClassifierExtension interface {
 }
 ```
 
-In the `cache/extensions` package, a host type for the extension point will be generated:
+In the `cache/customization` package, a host type for the extension point will be generated:
 
 ``` go
 // redis_extensions.go
 
-package extensions
+package customization
 
 type RedisExtensions struct{}
 ```
 
-This type doesn't contain any members as we require extension points to be pure functions; if they retain mutable state, we run the risk of introducing concurrency and sequence errors that would be hard to diagnose and fix.
+This type doesn't contain any members as we require extension points to be pure functions; if they contain mutable state, we run the risk of introducing concurrency and sequence errors that would be hard to diagnose and fix.
```

Comment:
For something like the vnet/subnet problem we're going to need to look up the subnets and put them into the vnet before sending to ARM, aren't we? That wouldn't be doable with a pure function.

Reply:
As long as the VNet is passed into the function, and back out again afterwards, I think it can.